File: frontend/src/services/workRecordService.ts
import { request } from '@/utils/request';

export interface WorkRecordCreateRequest {
  caseId: number;
  recordType: string;
  title: string;
  content: string;
  contactDate?: string;
  contactMethod?: string;
  contactResult?: string;
  contactDuration?: number;
  debtorResponse?: string;
  debtorAttitude?: string;
  nextAction?: string;
  nextContactDate?: string;
  followUpRequired?: boolean;
  isImportant?: boolean;
  templateId?: number;
  templateName?: string;
  attachments?: string;
  tags?: string;
  workDuration?: number;
  location?: string;
  weather?: string;
}

export interface WorkRecordResponse {
  id: number;
  caseId: number;
  handlerId: number;
  handlerName: string;
  recordType: string;
  recordTypeDesc: string;
  title: string;
  content: string;
  contactDate?: string;
  contactMethod?: string;
  contactMethodDesc?: string;
  contactResult?: string;
  contactResultDesc?: string;
  contactDuration?: number;
  formattedContactDuration?: string;
  debtorResponse?: string;
  debtorAttitude?: string;
  debtorAttitudeDesc?: string;
  nextAction?: string;
  nextContactDate?: string;
  followUpRequired?: boolean;
  isImportant?: boolean;
  templateId?: number;
  templateName?: string;
  attachments?: string;
  tags?: string;
  tagList?: string[];
  workDuration?: number;
  location?: string;
  weather?: string;
  createdAt: string;
  updatedAt: string;
  needsFollowUp?: boolean;
}

export interface WorkRecordQueryRequest {
  caseId?: number;
  handlerId?: number;
  recordType?: string;
  contactMethod?: string;
  contactResult?: string;
  debtorAttitude?: string;
  followUpRequired?: boolean;
  isImportant?: boolean;
  startDate?: string;
  endDate?: string;
  keyword?: string;
  tags?: string;
}

/**
 * 创建作业记录
 */
export const createWorkRecord = (data: WorkRecordCreateRequest) => {
  return request.post('/api/v1/work-records', data);
};

/**
 * 更新作业记录
 */
export const updateWorkRecord = (id: number, data: Partial<WorkRecordCreateRequest>) => {
  return request.put(`/api/v1/work-records/${id}`, data);
};

/**
 * 删除作业记录
 */
export const deleteWorkRecord = (id: number) => {
  return request.delete(`/api/v1/work-records/${id}`);
};

/**
 * 获取作业记录详情
 */
export const getWorkRecord = (id: number) => {
  return request.get<WorkRecordResponse>(`/api/v1/work-records/${id}`);
};

/**
 * 分页查询作业记录
 */
export const queryWorkRecords = (params: WorkRecordQueryRequest & {
  page?: number;
  size?: number;
}) => {
  return request.get('/api/v1/work-records', { params });
};

/**
 * 获取案件的所有作业记录
 */
export const getCaseWorkRecords = (caseId: number) => {
  return request.get<WorkRecordResponse[]>(`/api/v1/work-records/case/${caseId}`);
};

/**
 * 获取需要跟进的记录
 */
export const getFollowUpRecords = () => {
  return request.get<WorkRecordResponse[]>('/api/v1/work-records/follow-ups');
};

/**
 * 批量创建作业记录
 */
export const batchCreateWorkRecords = (data: WorkRecordCreateRequest[]) => {
  return request.post('/api/v1/work-records/batch', data);
};

/**
 * 获取作业记录统计
 */
export const getWorkRecordStatistics = (caseId?: number, handlerId?: number) => {
  return request.get('/api/v1/work-records/statistics', {
    params: { caseId, handlerId }
  });
};

File: backend/tmp/services_disabled/WorkRecordService.java
package com.drmp.service;

import com.drmp.dto.request.WorkRecordCreateRequest;
import com.drmp.dto.request.WorkRecordUpdateRequest;
import com.drmp.dto.request.WorkRecordQueryRequest;
import com.drmp.dto.response.PageResponse;
import com.drmp.dto.response.WorkRecordResponse;
import org.springframework.data.domain.Pageable;

import java.util.List;

/**
 * 作业记录服务接口
 *
 * @author DRMP Team
 * @since 1.0.0
 */
public interface WorkRecordService {

    /**
     * 创建作业记录
     */
    WorkRecordResponse createWorkRecord(WorkRecordCreateRequest request);

    /**
     * 更新作业记录
     */
    WorkRecordResponse updateWorkRecord(Long id, WorkRecordUpdateRequest request);

    /**
     * 删除作业记录（软删除）
     */
    void deleteWorkRecord(Long id);

    /**
     * 获取作业记录详情
     */
    WorkRecordResponse getWorkRecord(Long id);

    /**
     * 分页查询作业记录
     */
    PageResponse<WorkRecordResponse> queryWorkRecords(WorkRecordQueryRequest request, Pageable pageable);

    /**
     * 获取案件的所有作业记录
     */
    List<WorkRecordResponse> getCaseWorkRecords(Long caseId);

    /**
     * 获取需要跟进的记录
     */
    List<WorkRecordResponse> getFollowUpRecords();

    /**
     * 批量创建作业记录
     */
    List<WorkRecordResponse> batchCreateWorkRecords(List<WorkRecordCreateRequest> requests);

    /**
     * 获取作业记录统计
     */
    Object getWorkRecordStatistics(Long caseId, Long handlerId);
}